opengovsg/FormSG
# feat: add settings tab and toggle for multi lang feature

## Context
- Add settings tab and toggle to allow users to turn on the multi-language translation feature for their form.
- This PR is the second PR to introduce the settings tab and toggle view to allow users to turn on and off the multi-language translation feature.
- This PR does not introduce any changes to the public form and only introduces the functionality to turn on/off support for multi-language translation.

## Changes
- Add [views](https://www.figma.com/design/wNcuzzF3ciL8f6xYNvGZgK/Form-Exploration?node-id=15273-13197&t=hvNPQdzJyBb5NYwK-0) for settings tab and toggle.
- Add mutations to update the following form's setting field:
   - Set boolean value for `hasMultiLang` to indicate if a form supports the multiple languages
   - Set the languages, in `supportedLanguages` field,  that the a form supports

## Before & After Screenshots
| Description | Before | After | 
| ----------- | ------- | ----- |
| Addition of settings tab on the side bar and toggle for turning on/off support for multi-language translation feature| |<img width="1276" alt="Screenshot 2024-08-12 at 10 04 02 PM" src="https://github.com/user-attachments/assets/07467857-2da1-482e-ab23-73a2f1069fe0">|
| Language support list when toggle is turned on and success toast| |<img width="936" alt="Screenshot 2024-08-12 at 10 06 08 PM" src="https://github.com/user-attachments/assets/05ffc3ab-2d43-4051-a870-c0aab07398a3">|
| Toast when support for multi-translation is turned off| |<img width="763" alt="Screenshot 2024-08-12 at 10 20 23 PM" src="https://github.com/user-attachments/assets/7f827c30-9849-4996-9057-72ff5e730a41">|
| Tooltip when hovering over icon to able/disable language| |<img width="935" alt="Screenshot 2024-08-12 at 10 24 19 PM" src="https://github.com/user-attachments/assets/1117b129-63ad-460a-90d0-14e8b6f4aeda">|
| Tooltip when hovering over icon to edit translation| |<img width="914" alt="Screenshot 2024-08-12 at 10 21 05 PM" src="https://github.com/user-attachments/assets/aab4aa72-ca30-431b-ad4f-9db8ec1ccb1e">|
| Toast when disabling support for a specific language| |<img width="760" alt="Screenshot 2024-08-12 at 10 21 29 PM" src="https://github.com/user-attachments/assets/319ed96b-5769-447a-bb19-0d3d39f932d5">|
| Toast when enabling support for a specific language| |<img width="743" alt="Screenshot 2024-08-12 at 10 21 48 PM" src="https://github.com/user-attachments/assets/da72b75d-a365-443e-bb53-a5610b666a35">|

## Tests
TC 1: Turning on toggle adds support for default supported languages and sets the `hasMultiLang` attribute to be true
1. Turn on the toggle and check the forms table to verify that the `supportedLanguages` field is populated with the default supported languages (`[en-SG, zh-SG, ms-SG, ta-SG]`) and the `hasMultiLang` value is true

TC 2: Disable and then enable support for a specific language
1. Turn on toggle to add support for the multi-language translation feature
2. Disable "Chinese"
3. Verify in the forms table for the particular form that the `supportedLanguages` field now contains `[en-SG, ms-SG, ta-SG]`
4. Then enable "Chinese" language
5. Verify in the forms table for the particular form that the `supportedLanguages` field now contains `[en-SG, ms-SG, ta-SG]`
